### ISSUE TITLE
atexit()-based directory cleanup insufficient when runner is called via library interface

### ISSUE THREAD

**Issue by wenottingham** [ansible/ansible-runner#310]
Related to https://github.com/ansible/awx/issues/4010 - when runner is called as a python library, atexit() won't necessarily run at a point that it should.

**Comment by matburt** [comment]
I think the solution to this is to remove the `atexit()` call and use the system we use for directory isolation instead:

https://github.com/ansible/ansible-runner/blob/8691932005d78329383efe59357a397c7078807a/ansible_runner/runner.py#L221-L222

**Comment by ryanpetrello** [comment]
@matburt I don't think this is quite solved yet, seeing a lot of these types of errors in AWX integration:

![image](https://user-images.githubusercontent.com/214912/59108077-182f2f80-8908-11e9-8335-f197ac4deb6f.png)
